File: DSA/Sarthak_Jain_590014936/DAY_9/Day_9_Question_3.py
#Sarthak Jain 590014936
#Day 9 Question 3
#Find First Non-Repeating Character in a String

s = input("Enter a string: ")

freq = {}

# Count frequency of each character
for c in s:
    if c in freq:
        freq[c] += 1
    else:
        freq[c] = 1

# Find first character with frequency 1
found = False
for c in s:
    if freq[c] == 1:
        print("First non-repeating character:", "'" + c + "'")
        found = True
        break

if not found:
    print("-1")

File: DSA/AYUSHMEHTA_590015131/Day9_Question3.cpp
#include <iostream>
#include <unordered_map>

using namespace std;


char firstNonRepeatingChar(string &s){
  unordered_map<char, int> freq;
  for(char c:s){
      freq[c]++;
  }
  for(char c: s){
     if(freq[c]==1) return c;
  }
  return -1;

}


int main() {
    string s;
    cout << "Enter a string: ";
    getline(cin, s);
char result = firstNonRepeatingChar(s);
  if(result==-1){
        cout << "No non-repeating character found." << endl;
    
}
 else
        cout << "First non-repeating character: " << result << endl;
   

    return 0;
}
